source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java

```diff
@@ -770,7 +784,34 @@ public ExternalNote getExternalNoteByID(String zdbID) {
         return (ExternalNote) session.get(ExternalNote.class, zdbID);
     }
 
+    @Override
+    public List<DBLinkExternalNote> getDBLinkExternalNoteByDataZdbIDAndPublicationID(String dataZdbID, String publicationID) {
+        Session session = HibernateUtil.currentSession();
+        String hql = """
+                from DBLinkExternalNote
+                where externalDataZdbID = :dataZdbID
+                and publication.zdbID = :publicationID
+                """;
+        Query<DBLinkExternalNote> query = session.createQuery(hql);
+        query.setParameter("dataZdbID", dataZdbID);
+        query.setParameter("publicationID", publicationID);
+        return query.list();
+    }
+
+    @Override
+    public List<DBLinkExternalNote> getDBLinkExternalNoteByPublicationID(String publicationID) {

```

Comment:
You could consolidate those two methods and only have one, the one with the two args, do the Hibernate query. If the dataZdbID variable is null do not include it into the query.